tests/src/JIT/SIMD/Vector3Interop.cs

```diff
@@ -0,0 +1,339 @@
+    public static extern void nativeCall_RPInvoke_Vector3Arg(
+        CallBackDelegate_RPInvoke_Vector3Arg callBack);
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern bool nativeCall_RPInvoke_Vector3Ret(
+        CallBackDelegate_RPInvoke_Vector3Ret callBack);        
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3Array(
+        CallBackDelegate_RPInvoke_Vector3Array callBack, 
+        int v);
+           
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3InStruct(
+        CallBackDelegate_RPInvoke_Vector3InStruct callBack, 
+        int v);
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]        
+    public static extern bool nativeCall_RPInvoke_Vector3InComplexStruct(
+        CallBackDelegate_RPInvoke_Vector3InComplexStruct callBack);
+    
+    static bool result = false;
+   
+    static void callBack_RPInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2)
+    {
+        Vector3 tmp = new Vector3(2, 2, 2);
+        
+        // sum = (1, 2, 3) dot (2, 2, 2) = 12
+        float sum0 = Vector3.Dot(v1, tmp);   
+        // sum = (10, 20, 30) dot (2, 2, 2) = 20 + 40 + 60 = 120
+        float sum1 = Vector3.Dot(v2, tmp);        
+        
+        Console.WriteLine("callBack_RPInvoke_Vector3Arg:");
+        Console.WriteLine("    iVal {0}", i);
+        Console.WriteLine("    SumOfEles(v1) = {0} SumOfEles(v2) = {1}", sum0, sum1);
+        Console.WriteLine("    str {0}", s);
+        
+        result = (sum0 == 12) && (sum1 == 120) && (s == "abcdefg") && (i == 123);
+    }
+    
+    static Vector3 callBack_RPInvoke_Vector3Ret()
+    {
+        Vector3 tmp = new Vector3(1, 2, 3);
+        return tmp;
+    }        
+    
+    static void callBack_RPInvoke_Vector3Array(
+        [In, MarshalAs(UnmanagedType.LPArray, SizeParamIndex=1)] Vector3[] v, 
+        int size)
+    {   
+        Vector3 tmp = new Vector3(2, 2, 2);
+        
+        // sum0 = (2,3,4) dot (2,2,2) = 4 + 6 + 8 = 18
+        float sum0 = Vector3.Dot(v[0], tmp);
+        // sum0 = (11,21,31) dot (2,2,2) = 22 + 42 + 62 = 126
+        float sum1 = Vector3.Dot(v[1], tmp);
+        
+        Console.WriteLine("callBack_RPInvoke_Vector3Array: Sum0 = {0} Sum1 = {1}", sum0, sum1);
+        
+        result = (sum0 == 18) && (sum1 == 126);
+    }
+        
+    
+    static void callBack_RPInvoke_Vector3InStruct(DT v)
+    {   
+        Vector3 tmp = new Vector3(2, 2, 2);
+        
+        // sum0 = (2,3,4) dot (2,2,2) = 4 + 6 + 8 = 18
+        float sum0 = Vector3.Dot(v.a, tmp);
+        // sum1 = (11,21,31) dot (2,2,2) = 22 + 42 + 62 = 126
+        float sum1 = Vector3.Dot(v.b, tmp);
```

Comment:
Here too please compute the following dot products

v.a, v.b
v.a, v.a
v.b, v.b

I would expect this case to be handled correctly by RyuJIt, since a Vector3 is read off memory, RyuJIt will generate 8 byte followed by 4 byte memory read to load a Vector3 into xmm reg.